internal/pkg/db/memory/addressables_test.go

```diff
@@ -0,0 +1,737 @@
+			[]contract.Addressable{addressable2, addressable3},
+			false,
+			nil,
+		},
+		{
+			"No matching addressable",
+			allTestAddressables,
+			"non-existent publisher",
+			nil,
+			true,
+			db.ErrNotFound,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c, err := createAndPopulateClient(tt.prePopulatedAddressables)
+			if err != nil {
+				t.Errorf("Unexpected error during test setup: %v", err)
+				return
+			}
+
+			got, err := c.GetAddressablesByPublisher(tt.publisher)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetAddressablesByPublisher() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if tt.wantErr && err != nil && tt.errorType != nil {
+				if err.Error() != tt.errorType.Error() {
+					t.Errorf("Expected error of '%v', but got an error of  '%v'", tt.errorType, err)
+					return
+				}
+			}
+
+			if !tt.wantErr && !verifySliceEqualIgnoreOrder(got, tt.want) {
+				t.Errorf("GetAddressablesByPublisher() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestClient_DeleteAddressableById(t *testing.T) {
+	tests := []struct {
+		name                     string
+		prePopulatedAddressables []contract.Addressable
+		id                       string
+		wantErr                  bool
+		errorType                error
+	}{
+		{
+			"Delete existing addressable",
+			allTestAddressables,
+			addressable1.Id,
+			false,
+			nil,
+		},
+		{
+			"Delete non-existing addressable",
+			allTestAddressables,
+			"non-existing ID",
+			true,
+			db.ErrNotFound,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c, err := createAndPopulateClient(tt.prePopulatedAddressables)
+			if err != nil {
+				t.Errorf("Unexpected error during test setup: %v", err)
+				return
+			}
+
+			err = c.DeleteAddressableById(tt.id)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("DeleteAddressableById() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if tt.wantErr && err != nil && tt.errorType != nil {
+				if err.Error() != tt.errorType.Error() {
+					t.Errorf("Expected error of '%v', but got an error of  '%v'", tt.errorType, err)
+					return
+				}
+			}
+
+			if !tt.wantErr {
+				_, err := c.GetAddressableById(tt.id)
+				if err == nil || err.Error() != db.ErrNotFound.Error() {
+					t.Errorf("Expected the Addressable to be deleted but was not.")
+				}
+			}
+		})
+	}
+}
+
+// createAndPopulateClient constructs a new client and loads provided addressables into the underlying datastore.
+func createAndPopulateClient(addressables []contract.Addressable) (*Client, error) {
+	client := NewClient()
+
+	for _, addressable := range addressables {
+		_, err := client.AddAddressable(addressable)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return client, nil
+}
+
+// verifySliceEqualIgnoreOrder verfies the contents of two slices of Addressables are equal ignoring order.
+func verifySliceEqualIgnoreOrder(a, b []contract.Addressable) bool {
```

Comment:
I just wrote something roughly equivalent (but generic) for #2277.  Think about how you can write this generically and do so (so it can be reused by the other memory/db types' tests.  Generic version belongs in its own file.